### ISSUE TITLE
beginShape Vertex color doesn't work properly for non-triangle (or non-triangle-strip) shape types

### ISSUE THREAD

**Issue by redfood** [processing/p5.js#5269]
When using `beginShape` under `WEBGL` vertex coloring only works for `TRIANGLES` and `TRIANGLE_STRIP`.

#### Most appropriate sub-area of p5.js?

- [ ] Accessibility (Web Accessibility)
- [ ] Build tools and processes
- [x] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Friendly error system
- [ ] Image
- [ ] IO (Input/Output)
- [ ] Localization
- [ ] Math
- [ ] Unit Testing
- [ ] Typography
- [ ] Utilities
- [x] WebGL
- [ ] Other (specify if possible)

#### Details about the bug:

- p5.js version: **1.3.1**
- Web browser and version: **Chrome: Version 91.0.4472.77 (Official Build) (x86_64)**
- Operating System: **MacOS Big Sur version 11.3.1 (20E241)**
- Steps to reproduce this:

```
function setup() {
  createCanvas(400, 400, WEBGL);
}

function draw() {
  background(220);
  translate(-width / 2, -height / 2);

  beginShape(TRIANGLES);
 
    //t1
    fill(255,0,0); vertex(10,10);
    fill(0,255,0); vertex(10,200);
    fill(0,0,255); vertex(200,200);

    //t2
    fill(255,0,0); vertex(10,10);
    fill(0,0,255); vertex(200,200);
    fill(0,255,0); vertex(200,10);

  endShape(CLOSE);
  
  
  //This quad should look the same
  //except for the missing diagonal stroke.
 // but it has no fill
  beginShape(QUADS);
    fill(255,0,0); vertex(200,200);
    fill(0,255,0); vertex(200,390);
    fill(0,0,255); vertex(390,390);  
    fill(0,255,0); vertex(390,200);
  endShape(CLOSE);

}

```

**Comment by SanjaySinghRajpoot** [comment]
I went through the `beginShape` function in the WebGL folder (screenshot below check line 28). It is only made for Traingualar objects and not for non-triangular shapes. I think we need to add this feature for non-triangular objects, but I don't know how to do it. Can anyone help me out here?
![Screenshot from 2021-05-31 19-59-23](https://user-images.githubusercontent.com/67458417/120208601-49b70900-c24b-11eb-86d1-6867005513d4.png)

**Comment by davepagurek** [comment]
I believe this is fixed in 1.5.0!

![image](https://user-images.githubusercontent.com/5315059/214715896-209c0d8a-4e44-4543-986f-fc59d21fc497.png)


I'm going to close this issue for organizational purposes, but feel free to continue the discussion if you notice anything else or if this bug comes up again.
